### PR TITLE
feat: Add code for support repair kits

### PR DESCRIPTION
**Pull request by ChugunovRoman**

### CODE REVIEW

File: src/xrGame/ui/UIActorMenu.h [hunk 1]
```diff
@@ -258,6 +258,7 @@ class CUIActorMenu : public CUIDialogWnd, public CUIWndCallback
     bool AllowItemDrops(EDDListType from, EDDListType to);
 
     bool OnItemDrop(CUICellItem* itm);
+    bool OnItemDropOnItem(EDDListType t_old, EDDListType t_new, CUIDragDropListEx* old_owner, CUIDragDropListEx* new_owner);
```
Comment (Xottab-DUTY) [inline src/xrGame/ui/UIActorMenu.h]: Bad name. Nichego ne ponyatno...

Comment (ChugunovRoman) [reply src/xrGame/ui/UIActorMenu.h]: `DropItemOnAnotherItem` - tak norm?))

Comment (Xottab-DUTY) [reply src/xrGame/ui/UIActorMenu.h]: What this function does? :eyes:

Comment (ChugunovRoman) [reply src/xrGame/ui/UIActorMenu.h]: > What this function does? eyes

This function calls when player drag and drop an item on another item in a slot in the inventory or weapon/outfit slot. The function call Lua function and then code on Lua side determinate whats doing with these items.

File: src/xrGame/ui/UIActorMenu_action.cpp [hunk 2]
```diff
@@ -108,9 +139,16 @@ bool CUIActorMenu::OnItemDrop(CUICellItem* itm)
...
         if (CanSetItemToList(CurrentIItem(), new_owner, slot_to_place))
             ToSlot(itm, true, slot_to_place);
+
+        DropItemOnAnotherItem(t_old, t_new, old_owner, new_owner);
     }
     break;
-    case iActorBag: { ToBag(itm, true);
+    case iActorBag:
+    {
+        if(!DropItemOnAnotherItem(t_old, t_new, old_owner, new_owner))
```
Comment (Xottab-DUTY) [inline src/xrGame/ui/UIActorMenu_action.cpp]:
```suggestion
        if (!DropItemOnAnotherItem(t_old, t_new, old_owner, new_owner))
```

File: src/xrGame/ui/UIActorMenu_action.cpp [hunk 3]
```diff
@@ -69,12 +69,43 @@ void CUIActorMenu::OnDragItemOnTrash(CUIDragItem* item, bool b_receive)
...
+
+        CGameObject* GO1 = smart_cast<CGameObject*>(CurrentIItem());
+        CGameObject* GO2 = _iitem ? smart_cast<CGameObject*>(_iitem) : nullptr;
+        if (funct1(GO1 ? GO1->lua_game_object() : 0,
+            GO2 ? GO2->lua_game_object() : 0, (int)t_old, (int)t_new) == false)
+            return false;
+    }
+    //-Alundaio
+    return false;
+}
```
Comment (Xottab-DUTY) [inline src/xrGame/ui/UIActorMenu_action.cpp:105]:
```suggestion
}

```

**Review by Xottab-DUTY — APPROVED** [review]

**Comment by Xottab-DUTY** [comment]
Thank you!

**Pull request merged**